Training/Training/file.c

```diff
@@ -1,13 +1,133 @@
-// ------------------------------------------------------------------------------------------------
+// --------------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
+// Copyright (c) Gokul TS @Metamation India.
 // ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------------------
-
+// file.c
+// C Program to check test cases
+// --------------------------------------------------------------------------------------------------------
+#include "SortAndSearch.h"
 #include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdlib.h>
+#include <malloc.h>
+
+/// <summary>Gets integer input from user</summary>
+/// <param name="prompt"></param>
+int GetUserInput (char* prompt);
+/// <summary>Implements the search and sort algorithm</summary>
+void SortAndSearch (void(*func)(int[], int));
+/// <summary>Checks test cases</summary>
+void TestCases (void(*func)(int[], int), char* sortType);
+/// <summary>Returns whether the given array is sorted or not</summary>
+int IsSorted (int* arr1, int size);
+/// <summary>Clears the content on screen</summary>
+void ClrScreen ();
+
+int main () {
+   while (1) {
+      int num = GetUserInput ("1-BUBBLE SORT\n2-INSERTION SORT\n3-TEST CASES (B.S)"
+                              "\n4-TEST CASES (I.S)\n5-CLR SCREEN\n6-EXIT\nEnter an option:");
+      switch (num) {
+         case 1:
+            SortAndSearch (BubbleSort);
+            break;
+         case 2:
+            SortAndSearch (InsertionSort);
+            break;
+         case 3:
+            TestCases (BubbleSort, "Bubble Sort:     ");
+            break;
+         case 4:
+            TestCases (InsertionSort, "Insertion Sort:  ");
+            break;
+         case 5:
+            ClrScreen ();
+            break;
+         case 6:
+            return 0;
+         default:
+            printf (ANSI_COLOR_RED"Please enter a number between 1 - 6\n"ANSI_RESET_ALL);
+      }
+   }
+   return 0;
+}
+
+int GetUserInput (char* prompt) {
+   int number = 0;
+   for (;;) {
+      if (prompt != "") printf ("%s", prompt);
+      char strNum[MAX_DIGIT], * endPtr = NULL,
+         * strPtr = fgets (strNum, sizeof (strNum), stdin),
+         * validInput = strchr (strNum, '\n');
+      number = strtol (strNum, &endPtr, 10);
+      if (validInput == NULL)  while (getchar () != '\n');
+      if (strPtr == NULL || validInput == NULL || (!isdigit (strNum[0]) && strNum[0] != '-') || *endPtr != '\n') {
```

Comment:
```suggestion
      if (strPtr == NULL || (!isdigit (strNum[0]) && strNum[0] != '-') || *endPtr != '\n') {
```
You are already checking for this in the previous if statement. Keep one.